### PR TITLE
Preserve signed zero in FX complex codegen

### LINKED ISSUE #153852: [BUG] repr of complex tensors loses negative zero in imaginary part inside torch.compile
### 🐛 Describe the bug

Reproducer:

```python
import torch

def f(x):
    return x.cos(), x.__repr__()

x = torch.tensor([-1e-28j])
print(f(x))  # (tensor([1.-0.j]), 'tensor([-0.-1.0000e-28j])')
print(torch.compile(f, fullgraph=False, backend="eager")(x))  # (tensor([1.+0.j]), 'tensor([0.-1.0000e-28j])')
```

### Versions

Pytorch main branch

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* __->__ #185550

FX codegen used complex.__repr__ when rendering complex constants into generated Python source. CPython can print zero-component values such as (-0-1e-28j), -1e-28j, or (1-0j), and parsing that source can flip the sign of a zero component. Dynamo exposes this when it traces a tensor constant containing -1e-28j through FX: the generated GraphModule reconstructs a value with the wrong signed zero, changing tensor repr and the sign of the zero imaginary result from cos().

Render complex constants with a zero real or imaginary component as complex(real, imag) through the existing recursive argument printer. Float repr preserves -0.0, so those generated constants round-trip signed zero components. Nonzero complex constants keep the existing repr() path to avoid changing the common codegen case.

Fixes #153852

Generated by my agent

Benchmark Results:

- FX GraphModule construction with 1000 zero-component complex constants, 80 iterations x 7 repeats, median: main 3.55 ms/graph; this diff 6.77 ms/graph. This is the affected correctness path that now emits complex(real, imag) to preserve signed zero.

- FX GraphModule construction with 1000 nonzero complex constants, same command shape, median: main 4.29 ms/graph; this diff 3.90 ms/graph. Nonzero constants remain on the existing repr() path; the difference is measurement noise.

Test Plan:

- python test/dynamo/test_repros.py ReproTests.test_compile_complex_tensor_constant_signed_zero

- python test/test_fx.py TestFX.test_complex_constant_codegen_preserves_signed_zero (direct run blocked locally before target test by unrelated torchvision::nms registration failure)

- targeted TestFX.test_complex_constant_codegen_preserves_signed_zero via a torchvision import stub

- lintrunner -a

- git diff --check

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @jataylo @azahed98

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I marked this PR ready for review, requested @angelayi based on the FX graph code ownership, and reran the cancelled `before-test / llm-retrieval` job. The remaining inductor failure is classified by Dr. CI as broken trunk/unrelated, so I am waiting for the rerun to finish.

**Review by desertfire — APPROVED** [review]

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by jansel** [comment]
My agent says the mergebot failure was a Windows CI environment/setup build failure, with the remaining inductor failure classified by Dr. CI as broken trunk. I rebased the ghstack PR onto current main and resubmitted it so CI can rerun on a fresh base; the focused Dynamo regression, the targeted FX regression via the existing torchvision stub, and lintrunner all pass locally.

**Comment by jansel** [comment]
My agent says the remaining new CI failure was in the inductor torchbench training shard where eager itself produced nondeterministic accuracy results for mnasnet1_0, mobilenet_v2, and shufflenet_v2_x1_0; this does not appear related to the FX complex constant codegen change. I reran the failed inductor workflow jobs so Dr. CI can refresh.
